pubsub-mapped-api/src/main/java/com/google/pubsub/clients/producer/ExtendedConfig.java

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+package com.google.pubsub.clients.producer;
+
+import java.util.Map;
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerConfigAdapter;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.Range;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+
+/**
+ * Provides the configurations for a Kafka Producer instance.
+ */
+@VisibleForTesting
+public class ExtendedConfig {
```

Comment:
Let's be consistent in naming and layout with the consumer. That means renaming PubSubConfig to PubSubProducerConfig and ExtendedConfig to Config.

Reply:
I leave it to you and @pietrzykp to decide how you want to do it. If we aren't going to access the config in any way, then an inner class may be the best. But let's just make sure we are consistent across the Producer and Consumer.

Reply:
Inner class makes a lot of sense. Though, I feel like when user will be scanning through our files it would be easier for them to understand if it looks just as Kafka's configuration file they've already seen (like very-analogical, separate file). I'm fine with both ways.

Reply:
Also, my configuration is built entirely differently - in constructor it populates its attributes, so all of "consumerConfig.getString(ConsumerConfig.GROUP_ID_CONFIG)" are done there. It allows me not to keep this huge list of config attributes in Consumer implementation, rather then that I keep "config" object and just call config.getGroupId(). It also makes easier to distinguish between attributes that are configuration and different attributes of consumer.

Reply:
I think the user is not concerned about "our" files, he should be more concerned about any changes he will need to make - to his code -, which is the same in both cases. At the end of the day, they will just look at our documentation - which still, won't differ.

Reply:
Yeah, I have checked your configuration, it's elegant but I targeted a simple transition for now. It's on my TODO list.